service/storage_proxy.cc

```diff
@@ -421,6 +428,10 @@ class storage_proxy::remote {
         });
         auto& sp = _sp;
         co_await sp.mutate_counters_on_leader(std::move(mutations), cl, timeout, std::move(trace_state_ptr), /* FIXME: rpc should also pass a permit down to callbacks */ empty_service_permit());
```

Comment:
mutate_counters_on_leader() itself talks to replicas. Should we make sure it's using the same effective_replication_map as our caller?

Reply:
and should we use the same fencing token as was given to us?